provider/gandi/gandi.go

```diff
@@ -49,13 +49,63 @@ type GandiProvider struct {
 	DomainClient  DomainClientAdapter
 	domainFilter  endpoint.DomainFilter
 	DryRun        bool
+	TXTPrefix     string
+	TXTSuffix     string
 }
 
-func NewGandiProvider(ctx context.Context, domainFilter endpoint.DomainFilter, dryRun bool) (*GandiProvider, error) {
+const (
+	// RecordTypeALIAS is a RecordType enum value
+	RecordTypeALIAS = "ALIAS"
+)
+
+func getSupportedRecordTypes() []string {
+	return []string{
+		endpoint.RecordTypeA,
+		endpoint.RecordTypeAAAA,
+		endpoint.RecordTypeCNAME,
+		endpoint.RecordTypeSRV,
+		endpoint.RecordTypeTXT,
+		endpoint.RecordTypeNS,
+	}
+}
+
+func isRecordTypeSupported(recordType string) bool {
+	for _, t := range getSupportedRecordTypes() {
+		if t == recordType {
+			return true
+		}
+	}
+	return false
+}
+
+// dropAffix cleans-up the domain name by removing the prefix and suffix
+func dropAffix(RrsetName string, TXTPrefix string, TXTSuffix string, RecordType string) string {
+	cleanRrsetName := strings.TrimPrefix(RrsetName, TXTPrefix)
+	cleanRrsetName = strings.TrimSuffix(cleanRrsetName, TXTSuffix)
+
+	// New TXT records have a prefix containing the record type
+	if RecordType == endpoint.RecordTypeTXT {
+		for _, t := range getSupportedRecordTypes() {
+			recordTypePrefix := strings.ToLower(t) + "-"
+			cleanRrsetName = strings.TrimPrefix(cleanRrsetName, recordTypePrefix)
+		}
```

Comment:
It's checking "${prefix}-${record"
It's removing "${prefix}".
So resulting string will be like "-${record}".
=> It will never go to L178 of this file : 					`name = r.RrsetName`

I suggest to add a test case for this and, after, fixing it.